plugins/modules/mso_schema_site_contract_service_graph_listener.py

```diff
@@ -0,0 +1,744 @@
+
+            provider_epg_ref = rule.get("provider_epg_ref")
+            if provider_epg_ref:
+                rule_data["providerEpgRef"] = "/schemas/{0}/templates/{1}/anps/{2}/epgs/{3}".format(
+                    provider_epg_ref.get("schema") or schema_id,
+                    provider_epg_ref.get("template_name") or template,
+                    provider_epg_ref.get("anp_name"),
+                    provider_epg_ref.get("epg_name"),
+                )
+
+            health_check = rule.get("health_check")
+            if health_check:
+                health_check_data = dict()
+                if listener_protocol in ["tcp", "udp"]:
+                    if health_check.get("protocol") == "tcp":
+                        mso.input_validation("health_check - 'protocol'", "tcp", ["port", "unhealthy_threshold", "interval"], health_check)
+                    elif health_check.get("protocol") in ["http", "https"]:
+                        mso.input_validation("health_check - 'protocol'", "http/https", ["port", "path", "unhealthy_threshold", "interval"], health_check)
+                elif (listener_protocol == "http" and health_check.get("protocol") == "https") or (
+                    listener_protocol == "https" and health_check.get("protocol") == "http"
+                ):
+                    mso.input_validation(
+                        "health_check - 'protocol'", "http/https", ["port", "path", "unhealthy_threshold", "timeout", "interval"], health_check
+                    )
+                else:
+                    mso.fail_json(
+                        msg=(
+                            "The 'listener_protocol': {0} and the health_check protocol: {1} "
+                            + "is not a valid configuration at the object position: {2} and the object name: {3}"
+                        ).format(listener_protocol, health_check.get("protocol"), position, rule.get("name"))
+                    )
+
+                health_check_data["port"] = health_check.get("port")
+                health_check_data["protocol"] = health_check.get("protocol")
+                health_check_data["path"] = health_check.get("path")
+                health_check_data["interval"] = health_check.get("interval")
+                health_check_data["timeout"] = health_check.get("timeout")
+                health_check_data["unhealthyThreshold"] = health_check.get("unhealthy_threshold")
+                health_check_data["successCode"] = health_check.get("success_code")
+                health_check_data["useHostFromRule"] = YES_OR_NO_TO_BOOL_STRING_MAP.get(health_check.get("use_host_from_rule"))
+                health_check_data["host"] = health_check.get("host")
+
+                rule_data["healthCheck"] = health_check_data
+
+            rules_data.append(rule_data)
+
+        listener_object["rules"] = rules_data
+
+        if parent_present:
+            # Update an existing listener
+            if mso.existing:
+                listener_path = (
```

Comment:
Can't we determine the path during lookup already? Because there is already logic there that would determine where it exists if it exists.

Reply:
same applies for other assignments of listener_path